### DISCUSSION TITLE
View - Additional Discussion about Code Convention

### DISCUSSION THREAD

**Discussion by wherehows** [githru/githru-vscode-ext#99] [View]
1. 변수명으로 d를 쓰는게 맞을까?
https://github.com/githru/githru-vscode-ext/pull/94#discussion_r964328815

2. undefined 대신 null return하기
https://github.com/githru/githru-vscode-ext/discussions/14#discussioncomment-3410756
https://github.com/githru/githru-vscode-ext/pull/72#discussion_r962172200

3. 스타일 속성에 사이즈를 적용하지 않을 때 0을 기입할까 0px을 기입할까?
https://github.com/githru/githru-vscode-ext/pull/63#discussion_r961277127

4. px vs pt
https://github.com/githru/githru-vscode-ext/pull/63#discussion_r961277714

6. (Fixed) PR 올릴 때 어떠한 PR 인지 명시하기 (feat, enhance, chore)
https://github.com/githru/githru-vscode-ext/pull/88#issuecomment-1236824330

**Comment by jejecrunch** [comment]
3번과 관련하여 검색해보았는데요.
https://velog.io/@cks3066/CSS-0px-vs-0
이런 내용이 나왔습니다
그런데 0px를 0으로 기재했을 때 예외적인 상황이 어떤 상황이 있었는지에 대해서는 정확히 나와있지 않아 실험이 필요할 것 같습니다
우선 예외적인 상황이 있다는 거 자체가 0px를 0으로 기재하면 안되는 이유 중 하나일거라고 생각하고 저는 명시적으로 0px로 가는 것이 더 올바른 방향이라는 생각이 듭니다.

**Reply by wherehows** [reply]
사이드 이펙트가 언제 발생하는지는 잘 모르겠으나, stackoverflow만 보면 둘은 큰 차이가 없는 것 같네요. @__@ 통일성만 지켜주면 될거같아요.
https://stackoverflow.com/questions/4318471/property-0-or-property-0px-in-css

**Reply by wherehows** [reply]
4번과 관련해서 저도 좀 얕게 찾아봤는데요.
[W3C 아티클](https://www.w3.org/Style/Examples/007/units.en.html)에서는 고정값인 pt의 사용을 추천하지 않고 있긴하네요. 꽤 오래전에 작성된 문서이긴 합니다.

>The so-called absolute units (cm, mm, in, pt and pc) mean the same in CSS as everywhere else, but only if your output device has a high enough resolution. On a laser printer, 1cm should be exactly 1 centimeter. But on low-resolution devices, such as computer screens, CSS doesn't require that. And indeed, the result tends to be different from one device to another and from one CSS implementation to another. It's better to reserve these units for high-resolution devices and in particular for printed output. On computer screens and handheld devices, you'll probably not get what you expect.

>There is another reason to avoid absolute units for other uses than print: You look at different screens from different distances. 1cm on a desktop screen looks small. But the same on a mobile phone directly in front of your eyes looks big. It's better to use relative units, such as em, instead.

**Reply by jejecrunch** [reply]
자료 감사합니다 ! 그러면 em, rem으로 가는 게 맞지 않을까 하는 생각도 드네요.

**Reply by wherehows** [reply]
저는 rem을 보통 쓰는데,,,

rem이 몸이 불편하신 분들이 root 폰트 사이즈? 를 늘리면 그에따라 반응하니까 rem을 사용하는 것으로 알고있거든요.

그런데 vscode에서도 이를 고려해야할까...싶은... 너무 생각이 짧은걸까요

**Reply by jejecrunch** [reply]
> rem 은 최상위 태그의 font-size 값이 기준이며, em 은 현재 요소의 font-size 값이 기준이라는 것입니다.

cf. http://triki.net/prgm/3250

저는 rem을 쓰는 것이 좀 더 유지보수하기 편하다고 느껴지긴 합니다!

**Comment by jejecrunch** [comment]
https://github.com/githru/githru-vscode-ext/pull/157#discussion_r969127498

> 기존 entire, wrap, container 등과 같이 혼재하여 사용하던 네이밍을 모두 container로 통일해 두었습니다.
